### PR TITLE
Create deno.yml

### PR DESCRIPTION
**Pull request by johnperez416**

### CODE REVIEW

File: .github/workflows/deno.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# This workflow will install Deno and run tests across stable and nightly builds on Windows, Ubuntu and macOS.
+# For more information see: https://github.com/denolib/setup-deno
+
+name: Deno
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
```
Comment (johnperez416) [inline .github/workflows/deno.yml:15]: `[](url)`

File: .github/workflows/deno.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    strategy:
+      matrix:
+        deno: ["v1.x", "nightly"]
+        os: [macOS-latest, windows-latest, ubuntu-latest]
+
+    steps:
+      - name: Setup repo
+        uses: actions/checkout@v2
+
+      - name: Setup Deno
```
Comment (johnperez416) [inline .github/workflows/deno.yml:30]: - [ ]

File: .github/workflows/deno.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        uses: denolib/setup-deno@c7d7968ad4a59c159a777f79adddad6872ee8d96
+        with:
+          deno-version: ${{ matrix.deno }} # tests across multiple Deno versions
+
+      - name: Cache Dependencies
+        run: deno cache deps.ts
+
+      - name: Run Tests
+        run: deno test -A --unstable
```
Comment (johnperez416) [inline .github/workflows/deno.yml:39]:
```suggestion
        uses: denolib/setup-deno@c7d7968ad4a59c159a777f79adddad6872ee8d96
        with:
          deno-version: ${{ matrix.deno }} # tests across multiple Deno versions

      - name: Cache Dependencies
        run: deno cache deps.ts

      - name: Run Tests
        run: deno test -A --unstable
```

File: .github/workflows/deno.yml (hunk 2 above)
Comment (johnperez416) [inline .github/workflows/deno.yml:30]: <details>
<summary>Details</summary>



</details>

**Comment by gr2m** [comment]
Hey can you comment on what you are trying to achieve here? I’d appreciate if you could create an issue or a discussion before creating a pull request

**Pull request closed without merging**